File: solutions/go/src/playground/algorithm/longest_common_prefix.go
package algorithm

func longestCommonPrefix(strs []string) string {
	if len(strs) < 1 {
		return ""
	}

	i := -1
	for {
		if i+1 >= len(strs[0]) {
			break
		}

		key := strs[0][i+1]

		for j := 1; j < len(strs); j++ {
			if !(i+1 < len(strs[j]) && strs[j][i+1] == key) {
				goto T
			}
		}

		i++
	}

T:
	if i < 0 {
		return ""
	}
	return strs[0][:i+1]
}

File: solutions/java/src/main/java/playground/algorithm/LongestCommonPrefix.java
package playground.algorithm;

public class LongestCommonPrefix {

    public String longestCommonPrefix(String[] strs) {
        if (strs.length < 1) {
            return "";
        }

        int pos = -1;
        boolean interrupt = false;
        for (; !interrupt && pos + 1 < strs[0].length(); pos++) {
            char key = strs[0].charAt(pos + 1);
            for (int i = 1; i < strs.length; i++) {
                if (!(pos + 1 < strs[i].length() && strs[i].charAt(pos + 1) == key)) {
                    interrupt = true;
                    break;
                }
            }
        }

        if (pos < 0) {
            return "";
        }

        return strs[0].substring(0, pos);
    }
}
